server/oidc/index.js

```diff
@@ -14,17 +15,52 @@ const OIDCErrors = [
 ];
 const scope = ['openid', 'profile', 'email'].join(' ');
 
+const isSilentAuth = req => req.query.silent === 'true';
+const isResign = req => req.query.resign === 'true';
+const isLogoutRequest = req => req.query.action === 'logout';
+const isActiveStrategy = req => req.authStrategy === 'oidc';
+
+const getPromptParams = req => {
+  if (isResign(req)) return { prompt: 'login' };
+  if (isSilentAuth(req)) return { prompt: 'none' };
+  return {};
+};
+
+const getSilentAuthParams = req => {
+  const strategy = req.passport.strategy('oidc');
+  const callbackUri = new URL(strategy.options.callbackURL);
+  callbackUri.searchParams.set('silent', 'true');
+  return { redirect_uri: callbackUri.href };
+};
+
 const isOIDCError = err => OIDCErrors.some(Ctor => err instanceof Ctor);
 
 router
   .get('/', (req, res, next) => {
-    const isResign = req.query.resign === 'true';
-    const params = { scope, ...isResign && { prompt: 'login' } };
+    const strategy = req.passport.strategy('oidc');
+    if (isLogoutRequest(req)) return strategy.logout()(req, res, next);
+    const params = {
+      scope,
+      ...getPromptParams(req),
+      ...isSilentAuth(req) && getSilentAuthParams(req)
```

Comment:
Can this be changed so that `getSilentAuthParams` call `isSilentAuth` and then returning an empty object if it is not? If it can not then there should at least be brackets around this. It is not easy to follow what is happening and what has priority.